plugins/feeds/public/otx_alienvault.py

```diff
@@ -88,22 +88,37 @@ def analyze(self, item):
 
                 obs.tag(tags)
                 obs.add_context(self.name, context)
-
+                investigation.link_to(obs, "Observed", "OTXAlienVault")
             elif type_ind in entity.EntityType:
-                entity.Entity(
+                ent = entity.Entity(
                     name=otx_indic["indicator"],
                     type=self._TYPE_MAPPING.get(otx_indic["type"]),
                 ).save()
-
+                investigation.link_to(ent, "Observed", "OTXAlienVault")
             elif type_ind in indicator.IndicatorType:
                 if type_ind == indicator.IndicatorType.yara:
-                    indicator.Indicator(
-                        name=f"YARA_{otx_indic['indicator']}",
+                    # sometimes the content is empty
+                    if not otx_indic["content"]:
+                        continue
+                    r = None
+                    try:
+                        r = yara.compile(source=otx_indic["content"])
+                    except Exception as e:
+                        logging.error(f"Error compiling YARA rule: {e}")
+                        continue
+
+                    t = list(r)[0]
+                    ind_obj = indicator.Indicator(
+                        name=f"{t.identifier}",
                         pattern=otx_indic["content"],
                         type=indicator.IndicatorType.yara,
                         location="OTX",
                         diamond=indicator.DiamondModel.capability,
+                        description=t.meta["description"],
                     ).save()
```

Comment:
location=OTX is not how the location field was intended to be used. But given that it's an optional parameter anyways, I'd say to remove it from the indicator.

```suggestion
                    ind_obj = indicator.Indicator(
                        name=f"{t.identifier}",
                        pattern=otx_indic["content"],
                        type=indicator.IndicatorType.yara,
                        diamond=indicator.DiamondModel.capability,
                        description=t.meta["description"],
                    ).save()
```